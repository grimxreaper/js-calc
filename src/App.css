@import url('https://fonts.googleapis.com/css2?family=Balsamiq+Sans:wght@700&display=swap');

* {
  box-sizing: border-box;
}
body {
  margin: 0;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgb(0, 0, 0);

}

.result {
  height: 60px;
  background-color: rgb(232, 179, 245);
  width: 100%;
  border-radius: 5px;
}

.result p {
  font-size: 40px;
  margin: 5px;
  text-align: right;
  font-family: 'Balsamiq Sans', cursive;
  color: rgb(208, 248, 62)

}

.keypad {
  color:rgb(95, 36, 233);
  font-family: 'Balsamiq Sans', cursive;
}

.calculator-body {
  width: 250px;
  height: 400px;
  margin: auto;
  font-family: 'Balsamiq Sans', cursive;

}

.button {
  display: block;
  background-color: rgb(248, 62, 186);
  font-family: 'Balsamiq Sans', cursive;

}


button {
  width: 25%;
  height: 60px;
  font-size: 30px;
  font-family: 'Courier New', Courier, monospace;
  background-color: rgb(223, 200, 250);
  border-radius: 5px;
}

button:active {
  transform: scale(0.9);
}

button:hover {
  opacity: 0.75;

}
h1 {
  color: rgb(208, 248, 62);
  font-family: 'Balsamiq Sans', cursive;
  text-align: center;
  font-size: 40px;
}