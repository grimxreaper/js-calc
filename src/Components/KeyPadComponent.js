import React from "react";

const keypads = [
  { type: "operation", key: "(" },
  { type: "operation", key: "CE" },
  { type: "operation", key: ")" },
  { type: "operation", key: "AC" },
  { type: "operation", key: "/" },
  { type: "operation", key: "*" },
  { type: "operation", key: "-" },
  { type: "operation", key: "+" },
  { type: "key", key: 9 },
  { type: "key", key: 8 },
  { type: "key", key: 7 },
  { type: "key", key: 6 },
  { type: "key", key: 5 },
  { type: "key", key: 4 },
  { type: "key", key: 3 },
  { type: "key", key: 2 },
  { type: "key", key: 1 },
  { type: "key", key: 0 },
  { type: "operation", key: "=" },
  { type: "key", key: "." },
];

class KeyPadComponent extends React.Component {
  render() {
    return (
      <div className="button">
        {keypads.map((keypad) => {
          return (
            <button
              className="keypad"
              key={keypad.key}
              name={keypad.key}
              data-testid={keypad.key}
              onClick={(e) => this.props.onClick(keypad)}
            >
              {keypad.key}
            </button>
          );
        })}
      </div>
    );
  }
}

export default KeyPadComponent;
