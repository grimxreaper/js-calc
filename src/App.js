import React, { Component } from 'react';
import './App.css';
import ResultComponent from './Components/ResultComponent';
import KeyPadComponent from "./Components/KeyPadComponent";

class App extends React.Component {
  constructor(props){
    super(props)

    this.state = {
      result : ""
    }
  }
  onClick = (button) => {
    if (button === '='){
      this.calculate()
    }
    else if (button === 'AC'){
      this.reset()
    }
    else if (button === 'CE'){
      this.backspace()
    }
    else {
      this.setState({
        result: this.state.result + button
      })
    }
  }

  calculate = () => {
    let currentResult = ""

    if (this.state.result === '--') {
      currentResult = this.state.result.replace('--', '+')
    }
    else {
      currentResult = this.state.result
    }
    try {
      this.setState({
        result: (eval(currentResult) || "") + ""
      })
    }
    catch (e){
      this.setState({
        result: "error"
      })
    }
  }

  reset = () => {
    this.setState({
      result: ""
    })
  }

  backspace = () => {
    this.setState({
      result: this.state.result.slice(0, -1)
    })
  }

  render(){
    return (
      <div >
        <h1>Pocket Js Calculator</h1>
        <div className="calculator-body">
          <ResultComponent result = {this.state.result}/>
          <KeyPadComponent onClick = {this.onClick}/>
        </div>
      </div>
    )
  }

}
export default App;