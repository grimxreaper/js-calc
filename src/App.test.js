import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import App from './App';
import KeyPadComponent from './Components/KeyPadComponent';
import ResultComponent from './Components/ResultComponent';


afterEach(cleanup)
//snapshot
it("matches snapshot", () => {
  const { asFragment } = render(<App />)
  expect(asFragment(<App />)).toMatchSnapshot()
});

//Testing DOM elements
it('should equal an empty string', () => {
  const { getByTestId } = render(<ResultComponent />);
  expect(getByTestId('result')).toHaveTextContent('')
});

//Testing events
it('displays open parens when clicked', () => {
  const { getByTestId } = render(<App />);
  
  fireEvent.click(getByTestId("("))

  expect(getByTestId("result")).toHaveTextContent("(")

})