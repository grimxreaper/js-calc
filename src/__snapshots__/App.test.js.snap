// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Pocket Js Calculator
    </h1>
    <div
      class="calculator-body"
    >
      <div
        class="result"
        data-testid="result"
      >
        <p />
      </div>
      <div
        class="button"
      >
        <button
          class="keypad"
          data-testid="("
          name="("
        >
          (
        </button>
        <button
          class="keypad"
          data-testid="CE"
          name="CE"
        >
          CE
        </button>
        <button
          class="keypad"
          data-testid=")"
          name=")"
        >
          )
        </button>
        <button
          class="keypad"
          data-testid="AC"
          name="AC"
        >
          AC
        </button>
        <button
          class="keypad"
          data-testid="/"
          name="/"
        >
          /
        </button>
        <button
          class="keypad"
          data-testid="*"
          name="*"
        >
          *
        </button>
        <button
          class="keypad"
          data-testid="-"
          name="-"
        >
          -
        </button>
        <button
          class="keypad"
          data-testid="+"
          name="+"
        >
          +
        </button>
        <button
          class="keypad"
          data-testid="9"
          name="9"
        >
          9
        </button>
        <button
          class="keypad"
          data-testid="8"
          name="8"
        >
          8
        </button>
        <button
          class="keypad"
          data-testid="7"
          name="7"
        >
          7
        </button>
        <button
          class="keypad"
          data-testid="6"
          name="6"
        >
          6
        </button>
        <button
          class="keypad"
          data-testid="5"
          name="5"
        >
          5
        </button>
        <button
          class="keypad"
          data-testid="4"
          name="4"
        >
          4
        </button>
        <button
          class="keypad"
          data-testid="3"
          name="3"
        >
          3
        </button>
        <button
          class="keypad"
          data-testid="2"
          name="2"
        >
          2
        </button>
        <button
          class="keypad"
          data-testid="1"
          name="1"
        >
          1
        </button>
        <button
          class="keypad"
          data-testid="0"
          name="0"
        >
          0
        </button>
        <button
          class="keypad"
          data-testid="="
          name="="
        >
          =
        </button>
        <button
          class="keypad"
          data-testid="."
          name="."
        >
          .
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
